rir/src/compiler/debugging/stream_logger.h

```diff
@@ -0,0 +1,63 @@
+#ifndef PIR_LOGGER_H
+#define PIR_LOGGER_H
+
+#include "../../runtime/Function.h"
+#include "../pir/pir_impl.h"
+#include "debugging.h"
+//#include "../translations/pir_2_rir.h"
+
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <stack>
+//#include <fstream>
+
+namespace rir {
+namespace pir {
+
+class StreamLogger {
+  public:
+    StreamLogger(DebugOptions options) : options(options) {}
+    ~StreamLogger() {
+        for (std::map<rir::Function*, std::stringstream*>::iterator it =
+                 streams.begin();
+             it != streams.end(); ++it) {
+            this->finish(it->first, *streams.at(it->first));
+            /*if (options.includes(DebugFlag::PrintIntoFiles)) {
+                std::ofstream outFile;
+                outFile.open(".pir/pirCompilation.data");
+                outFile << stream.str();
+                outFile.close();
+            } else {*/
+            std::cout << it->second->str();
+            //}
+            delete it->second;
+        }
+    }
+
+    void startLogging(rir::Function* function);
+    void endLogging();
+    void compilationInit();
+    void compilationEarlyPir(Closure&);
+    void pirOptimizations(Closure&, const std::string&, const std::string&,
+                          size_t);
+    void phiInsertion(Closure& closure);
+    void afterCSSA(Code*);
+    // void afterLiveness(SSAAllocator&);
```

Comment:
So the problem here is that it is kind of a mess to pass the SSAAllocator to the logger because that type is defined inside pir_2_rir but without a namespace. One alternative is to change the definition. But I do not think we will need to use this debug information yet. 
I left the comments so that it is explicit that it got commented now and we can uncommented it when someone finds the necessity to debug liveness.

Reply:
In general, I don't like commented out code, because it's not clear what it's for. And it often diverges from the rest of the code, since it's not executed or tested.

As a temporary measure, could you leave a comment?